x-pack/spec/config_management/elasticsearch_source_spec.rb

```diff
@@ -114,10 +116,45 @@
           #"xpack.management.elasticsearch.password" => elasticsearch_password,
         }
       end
+
       it "should raise an ArgumentError" do
         expect { described_class.new(system_settings) }.to raise_error(ArgumentError)
       end
     end
+
+    context "cloud settings" do
+      let(:cloud_name) { 'abcdefghijklmnopqrstuvxyz' }
+      let(:cloud_domain) { 'elastic.co' }
+      let(:cloud_id) { "label:#{Base64.urlsafe_encode64("#{cloud_domain}$#{cloud_name}$ignored")}" }
+
+      let(:settings) do
+        {
+            "xpack.management.enabled" => true,
+            "xpack.management.pipeline.id" => "main",
+            "xpack.management.elasticsearch.cloud_id" => cloud_id,
+            "xpack.management.elasticsearch.cloud_auth" => "#{elasticsearch_username}:#{elasticsearch_password}"
+        }
+      end
+
+      it "should not raise an ArgumentError" do
+        expect { described_class.new(system_settings) }.not_to raise_error
+      end
+
+      context "when cloud_auth isn't set" do
+        let(:settings) do
+          {
+              "xpack.management.enabled" => true,
+              "xpack.management.pipeline.id" => "main",
+              "xpack.management.elasticsearch.cloud_id" => cloud_id,
+              #"xpack.management.elasticsearch.cloud_auth" => "#{elasticsearch_username}:#{elasticsearch_password}"
```

Comment:
It is need or not in the test?

Reply:
intended: just a hint what is missing from the configuration
(its the same 2 specs above where a missing password is tested)